### PR TITLE
Feature/reset password

### PR DESCRIPTION
**Pull request by PavloJedi**

### CODE REVIEW

File: back-end/controllers/UserController.js [hunk 1]
```diff
@@ -1,4 +1,7 @@
-const { findByEmail, updateByID } = require('../services/UserService');
+const jwt = require('jsonwebtoken');
```
Comment (andriydzhrvts) [inline back-end/controllers/UserController.js]: Unused import

File: back-end/controllers/UserController.js [hunk 2]
```diff
@@ -23,4 +26,24 @@ const updateUserById = async (req, res) => {
   }
 };
 
-module.exports = { getUser, updateUserById };
+const changePassword = async (req, res) => {
+  try {
+    const { currentPassword, newPassword } = req.body;
+    const { id } = req.params;
+    const userPass = await getCurrentPassword(id);
+    const isValidPassword = await bcrypt.compare(currentPassword, userPass.password);
```
Comment (andriydzhrvts) [inline back-end/controllers/UserController.js]: If we use this method few times in different places in our project, we could create a new service with bcrypt for compare and hash password.

File: back-end/controllers/UserController.js [hunk 3]
```diff
@@ -23,4 +26,24 @@ const updateUserById = async (req, res) => {
   }
 };
 
-module.exports = { getUser, updateUserById };
+const changePassword = async (req, res) => {
+  try {
+    const { currentPassword, newPassword } = req.body;
+    const { id } = req.params;
+    const userPass = await getCurrentPassword(id);
```
Comment (andriydzhrvts) [inline back-end/controllers/UserController.js:31]: You could get all user object and avoid this method

File: back-end/controllers/UserController.js (hunk 2 above)
Comment (PavloJedi) [reply back-end/controllers/UserController.js]: created a new service with bcrypt for compare and hash password

File: back-end/controllers/UserController.js (hunk 1 above)
Comment (PavloJedi) [reply back-end/controllers/UserController.js]: deleted unused import

File: back-end/controllers/UserController.js [hunk 4]
```diff
@@ -23,4 +24,25 @@ const updateUserById = async (req, res) => {
...
+    const isValidPassword = await comparePasswords(currentPassword, userPass.password);
+    if (!isValidPassword) {
+      return res
+        .status(400)
+        .json({ status: 'error', message: 'Please enter correct old password' });
+    }
+    const hashedPass = await hashPasswords(newPassword, 10);
+    console.log(hashedPass);
```
Comment (andriydzhrvts) [inline back-end/controllers/UserController.js]: Remove console log

File: back-end/services/AuthService.js [hunk 5]
```diff
@@ -45,4 +45,12 @@ const comparePasswords = (pass1, pass2) =>
     });
   });
 
-module.exports = { register, login };
+const hashPasswords = (pass) =>
```
Comment (andriydzhrvts) [inline back-end/services/AuthService.js]: hashPassword

File: back-end/services/AuthService.js [hunk 6]
```diff
@@ -2,8 +2,8 @@ const User = require('../models/User');
 const bcrypt = require('bcryptjs');
 const jwt = require('jsonwebtoken');
 
-const register = (data) => {
-  bcrypt.hash(data.password, 10, (err, hashedPass) => {
+const register = async (data) => {
+  await hashPasswords(data.password, 10, (err, hashedPass) => {
```
Comment (andriydzhrvts) [inline back-end/services/AuthService.js]: Are you sure that this function will be work? Function `hashPasswords` needed 1 parameter, but you pass 3.

File: back-end/services/AuthService.js (hunk 5 above)
Comment (PavloJedi) [reply back-end/services/AuthService.js]: renamed to hashPassword

File: back-end/services/AuthService.js (hunk 6 above)
Comment (PavloJedi) [reply back-end/services/AuthService.js]: it works properly, tried it with registration

File: back-end/controllers/UserController.js (hunk 4 above)
Comment (PavloJedi) [reply back-end/controllers/UserController.js]: removed

**Review by andriydzhrvts — APPROVED** [review]

**Pull request merged**